src/plugins/query_enhancements/server/utils/facet.ts

```diff
@@ -43,6 +43,7 @@ export class Facet {
       const { format, lang } = request.body;
       const params = {
         body: {
+          fetch_size: request.body.fetch_size,
```

Comment:
if we have the context still here are we not able to just access advanced settings in this file and set the param? or was there a technical blocker that we should specify for each strategy the size if set or not.

and should we check if this value is set or not if it is then we pass it? what are implications of passing undefined to this the API? because right now i believe it will fire a request like

```
{
  body: {
    fetch_size: undefined,
    query: query.query,
    // ... other properties
  },
  // ... format property if applicable
}
```

will it be ignored?

Reply:
Quick thought - was worried about async search strategies (like S3), but I think doing it here in the fetch call should actually cover everything. Since async searches eventually hit this fetch method anyway, we should be good.

Mind double-checking that though? Just want to make sure I'm not missing something with the async flow

Reply:
directly passing the fetch size in the facet will cause async query to fail since it doesn't accept fetch_size as a param
```
const fetchSize = await context.core.uiSettings.client.get('discover:sampleSize');
      const params = {
        body: {
          fetch_size: fetchSize,
          query: query.query,
          ...(meta?.name && { datasource: meta.name }),
          ...(meta?.sessionId && {
            sessionId: meta.sessionId,
          }),
          ...(lang && { lang }),
        },
        ...(format !== 'jdbc' && { format }),
      };
```

<img width="1610" alt="Screenshot 2025-02-12 at 5 51 14 PM" src="https://github.com/user-attachments/assets/77978310-7174-4882-a6a4-90c0450e8a37" />

Reply:
i think since async search for s3 and PPL doesn't support fetch_size yet, we can interpolate the default query string for s3 dataset types and PPL queries for now. The limit number we get from advanced setting

`source = ${dataset.title} | head 500`

Reply:
Will change to only pass in fetch_size if fetch_size is defined.